### ISSUE TITLE
🐛 `lint/noRedeclare` should allow type param and function param with the same name

### ISSUE THREAD

**Issue by Conaclos** [biomejs/biome#1992]
### Environment information

```block
Playground
```

### What happened?

Using the same name for a function parameter and a type parameter triggers `lint/noRedeclare`.
[Playground](https://biomejs.dev/playground/?lintRules=all&code=ZQB4AHAAbwByAHQAIABmAHUAbgBjAHQAaQBvAG4AIABmADwAYQA%2BACgAYQA6ACAAYQApACAAewAKACAAIAByAGUAdAB1AHIAbgAgAGEAOwAKAH0A).

### Expected result

Because a type parameter is a type and a function parameter is a variable, both should coexist. This should not be reported as a redeclaration.

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct
